docs/subnets/create-a-simple-rust-vm.md

```diff
@@ -0,0 +1,1256 @@
+        Self {
+            parent_id: ids::Id::empty(),
+            height: 0,
+            timestamp: 0,
+            data: Vec::new(),
+
+            status: choices::status::Status::default(),
+            bytes: Vec::new(),
+            id: ids::Id::empty(),
+
+            state: state::State::default(),
+        }
+    }
+
+    pub fn new(
+        parent_id: ids::Id,
+        height: u64,
+        timestamp: u64,
+        data: Vec<u8>,
+        status: choices::status::Status,
+    ) -> io::Result<Self> {
+        let mut b = Self {
+            parent_id,
+            height,
+            timestamp,
+            data,
+            ..Default::default()
+        };
+
+        b.status = status;
+        b.bytes = b.to_slice()?;
+        b.id = ids::Id::sha256(&b.bytes);
+
+        Ok(b)
+    }
+
+    pub fn to_json_string(&self) -> io::Result<String> {
+        serde_json::to_string(&self).map_err(|e| {
+            Error::new(
+                ErrorKind::Other,
+                format!("failed to serialize Block to JSON string {}", e),
+            )
+        })
+    }
+
+    /// Encodes the Block to JSON in bytes.
+    pub fn to_slice(&self) -> io::Result<Vec<u8>> {
+        serde_json::to_vec(&self).map_err(|e| {
+            Error::new(
+                ErrorKind::Other,
+                format!("failed to serialize Block to JSON bytes {}", e),
+            )
+        })
+    }
+
+    /// Loads Block from JSON bytes.
+    pub fn from_slice(d: impl AsRef<[u8]>) -> io::Result<Self> {
+        let dd = d.as_ref();
+        let mut b: Self = serde_json::from_slice(dd).map_err(|e| {
+            Error::new(
+                ErrorKind::Other,
+                format!("failed to deserialize Block from JSON {}", e),
+            )
+        })?;
+
+        b.bytes = dd.to_vec();
+        b.id = ids::Id::sha256(&b.bytes);
+
+        Ok(b)
+    }
+
+    /// Returns the parent block Id.
+    pub fn parent_id(&self) -> ids::Id {
+        self.parent_id
+    }
+
+    /// Returns the height of this block.
+    pub fn height(&self) -> u64 {
+        self.height
+    }
+
+    /// Returns the timestamp of this block.
+    pub fn timestamp(&self) -> u64 {
+        self.timestamp
+    }
+
+    /// Returns the data of this block.
+    pub fn data(&self) -> &[u8] {
+        &self.data
+    }
+
+    /// Returns the status of this block.
+    pub fn status(&self) -> choices::status::Status {
+        self.status.clone()
+    }
+
+    /// Updates the status of this block.
+    pub fn set_status(&mut self, status: choices::status::Status) {
+        self.status = status;
+    }
+
+    pub fn bytes(&self) -> &[u8] {
+        &self.bytes
+    }
+
+    pub fn id(&self) -> ids::Id {
+        self.id
+    }
+
+    pub fn set_state(&mut self, state: state::State) {
+        self.state = state;
+    }
+
+    /// Verifies Block properties (e.g., heights),
+    /// and once verified, records it to the State.
+    pub async fn verify(&mut self) -> io::Result<()> {
```

Comment:
Just a question, do we need to show the following functions twice in this doc?
* verify
* accept
* reject

as each of these has a section below dedicated to them. If we need both, then code/comments should match exactly. Currently they are not exact the same.

Reply:
yeah going to update block for now then will have a followup PR that simplifies this and gets into more details of the implementation.